Feature: Changed Code Editor to Monaco

BanceDev opened the pull request:
Adds code to embed the Monaco editor in place of the ace editor. Also removes old Ace editor code. Note: This PR does not include a diverse selection of editor themes like previously offered for Ace. I would be more than happy to make a follow up PR to this to add more options however.

BanceDev commented:
if you want me to add the code that I am pulling from the CDN to the public/js directory I can do that so its in line with other external packages like ace used to be

kyledrake commented:
Oh good lord, I thought you were talking about the Monaco _font_ (as in, the literal font designed by Susan Kare) not the code editor. I'll take a look but this could be a tougher thing to transition from.

BanceDev commented:
I've also been trying to hunt it down. I will let you know if I find it.

BanceDev commented:
> Oh good lord, I thought you were talking about the Monaco _font_ (as in, the literal font designed by Susan Kare) not the code editor. I'll take a look but this could be a tougher thing to transition from.

Hahaha. Yeah it's a bit of a shift. If it's not worth doing it's all good. I just think it might be a nice upgrade.

kyledrake commented:
I think you're right, it's worth doing. The main issues here: fixing the styling, seeing if the syntax colors can be added back in.

kyledrake commented:
I figured out a source for themes (https://github.com/brijeshb42/monaco-themes). I suppose I could disable the autocomplete for now since it's new functionality anyways. This change also breaks the syntax error detection, but I'm sure it won't be hard to figure that out. I also need to figure out how to load the files locally (for security reasons I try to not run assets from CDNs) but again quite solvable.

It's probably at a good enough state that I can merge this in and try to run with the football and hopefully get it out by end of week. I'm approaching my end of day here but I'll pick this up again tomorrow. If you figure out the style stuff LMK.

BanceDev commented:
I'll see what I can do about the styles and adding back in error/warning checks. I'll also mess around with hosting the js instead of using the cdn and see what happens.

BanceDev commented:
I added an semi-hacky fix for the autocomplete suggestion styling. It seems like some kind of styling is being attributed to span elements at some higher order but I couldn't find it so I just added custom styles in the style tag in the .erb file.

kyledrake commented:
I'm going to merge this and see if I can't get the rest of it figured out. Thank you!
